sdk/src/rustls/common.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+use crate::Error;
+use rustls::Certificate;
+use rustls_pemfile::{read_one, Item};
+use std::{
+    io::{self, BufReader},
+    iter,
+};
+
+pub struct AlgorithmData {
+    pub rustls_id: rustls::SignatureScheme,
+    pub verification_alg: &'static dyn ring::signature::VerificationAlgorithm,
+    // This is because the Ring verification algorithm expects the key in a SubjectPublicKey, Openssl gives it in a SubjectPublicKeyInfo.
+    // We can ask Openssl to give us that with RSAPublicKey_out, but that's (i) only for RSA, (ii) not what happens in existing tests.
+    // https://github.com/briansmith/ring/issues/881#issuecomment-592749266
+    pub spk_offset: usize,
```

Comment:
Flagging that hack here. As the structures are well defined, it should be safe to just fast forward to the bytes where the usable part is.

Still.

Reply:
Removed